lib/retriable.rb

```diff
@@ -55,14 +55,10 @@ def retriable(opts = {})
         end
       rescue *[*exception_list] => exception
         if on.kind_of?(Hash)
-          patterns = [*on[exception.class]]
-          message_match = patterns.empty? ? true : false
-          patterns.each do |pattern|
-            if !!(exception.message =~ pattern)
-              message_match = true
-              break
-            end
+          message_match = exception_list.select { |e| exception.is_a?(e) }.inject(false) do |match, e|
+            match || [*on[e]].empty? || [*on[e]].any? { |pattern| exception.message =~ pattern }
```

Comment:
that's true though it is only going to check for exceptions that have a matched class (`.select { |e| exception.is_a?(e) }`) - so it will only have multiple items to travers if you are retrying multiple exceptions that are subclasses of each other.

whether it matters i guess depends on how optimized you care about a super fast ms order of magnitude list iteration in the context of a exception handling retry that is probably going to wait at least a second and often more.